src/service.rs

```diff
@@ -131,6 +131,21 @@ impl GrpcRequest {
 }
 
 pub type Headers = Vec<(String, String)>;
+#[derive(Debug)]
+pub enum HeaderKind {
+    Response(Headers),
+    Request(Headers),
+}
+
+impl HeaderKind {
+    pub fn is_empty(&self) -> bool {
+        match self {
+            HeaderKind::Response(headers) => headers.is_empty(),
+            HeaderKind::Request(headers) => headers.is_empty(),
+        }
```

Comment:
You don't need to change this, but if you were wondering how to do it in one liner: 
```suggestion
        match self {
            HeaderKind::Response(headers) | HeaderKind::Request(headers) => headers.is_empty(),
        }

        }
```

Reply:
Oh yeah forgot about that, will add that as it's nicer